### PR TITLE
[ADD] module 'mrp_bom_hierarchy'.

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This module was written to extend the functionality of Bill of
Materials to support users to better maintain the BOM hierarchy.

This module adds a new menu entry 'Bill of Materials Hierarchy' in
'Manufacturing | Products', from which the user can create a complete BOM
hierarchy from the tree view.

The user can now search using the field 'Full BOM Hierarchy Reference' to find
all the BOM hierarchy associated to a particular BOM Reference at once.

The user can now search using the field 'Full BOM Hierarchy Name' to find
all the BOM hierarchy associated to a particular BOM Name at once.

### CODE REVIEW

File: mrp_bom_hierarchy/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+This module adds a new menu entry 'Bill of Materials Hierarchy' in
+'Manufacturing | Products', from which the user can create a complete BOM
+hierarchy from the tree view.
+
+The user can now search using the field 'Full BOM Hierarchy Reference' to find
+all the BOM hierarchy associated to a particular BOM Reference at once.
+
+The user can now search using the field 'Full BOM Hierarchy Name' to find
+all the BOM hierarchy associated to a particular BOM Name at once.
+
```
Comment (bealdav) [inline mrp_bom_hierarchy/__openerp__.py:47]: Maybe easier to read with:
The user can now search using the field 'Full BOM Hierarchy Reference' (or Name) to find
all the BOM hierarchy associated to a particular BOM Reference (or Name) at once.

File: mrp_bom_hierarchy/model/mrp.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+            while b:
+                if b.code:
+                    data.insert(0, b.code)
+                else:
+                    data.insert(0, '')
+
+                b = b.bom_id
+            data = ' / '.join(data)
+            data = '[' + data + '] '
+
```
Comment (bealdav) [inline mrp_bom_hierarchy/model/mrp.py:114]: Better performance in a loop with 

```
data = '[%s] ' % data
```

**Comment by JordiBForgeFlow** [comment]
Thanks @bealdav . I have now included your review comments.

**Comment by JordiBForgeFlow** [comment]
I'm closing this PR now.

**Pull request closed without merging**